### DISCUSSION TITLE
Third party dependencies

### DISCUSSION THREAD

**Discussion by AdriandMartin** [zellij-org/zellij#2430] [Q&A] [answered]
I'm a newbie with terminal multiplexers, but I have recently discovered this project and seems to be amazing. I knew that both tmux and screen use ncurses to manage the terminal output, but zellij doesn't, so how do you implement it? Does you use your own libraries for that? Thank you

**Comment by Lockszmith-GH** [comment]
Look at [Cargo.toml](https://github.com/zellij-org/zellij/blob/main/Cargo.toml) that's the list of libraries used.

In `rust` everything is statically linked be default - so you get a single binary file for easy distribution.

**Reply by Lockszmith-GH** [reply]
Looks like [dialoguer](https://docs.rs/dialoguer/0.7.1/dialoguer/index.html) is the library responsible for rendering on the console.

**Reply by AdriandMartin** [reply]
Thanks, I'm not used to working with Rust and I didn't know how the projects are built. It was exactly what I was looking for :)

**Reply by tlinford** [reply, marked as answer]
I'm pretty sure dialoguer is only used for some minor stuff, and the rendering is custom.
Imsnif wrote a nice article a while back to explain "the basics" of some of this stuff: https://poor.dev/blog/terminal-anatomy/

**Reply by AdriandMartin** [reply]
The post is great! Thank you for the clarification
